### DISCUSSION TITLE
Understanding the better performance of CPSAT 9.7 over 9.10 on my problem

### DISCUSSION THREAD

**Discussion by Sacropyne** [google/or-tools#4310] [CP-SAT questions]
Hi,

I recently tried a scheduling problem mainly characterized by NoOverlap constraints, along with some linear constraints on the sum of some interval sizes, plus one Cumulative constraint. I had two Python environements available on my server, one with OrTools 9.10.4067 and one with 9.7.2996, and without realizing I was doing it, I ran the same satisfaction problem on the first and then the second, and while 9.7 proves infeasibility almost instantaneously, 9.10 hits the 100 seconds timeout and exits with an unknown status. My guess is that there is nothing 9.7 can do that 9.10 can’t, so it must be a question of default parameter tuning. I have joined the two run logs below, are they the right way to try to understand the cause of the performance mismatch and if yes can you help me decoding them ?

Log 9.10 : https://drive.google.com/file/d/1Rgwh81kwyJILWEWlP2HAxmxjrGV-aOxS/view?usp=drivesdk

Log 9.7 : https://drive.google.com/file/d/1lk2gmVs2PuDY2dJVlrPiPea8SuPPu9Nq/view?usp=drivesdk

Thanks a lot !

**Comment by lperron** [comment]
Maybe 9.7 is wrong. 

Let 9.10 run overnight

**Reply by Sacropyne** [reply]
Hi,

Thanks !

Here is the night run : 9.10 returns Unknown status after 45000 seconds of search 

https://drive.google.com/file/d/190mcd4CEmbOFmP2TgsfT3m5Jsk_4sAXe/view?usp=drivesdk

What would you suggest to understand this further ?

**Comment by lperron** [comment]
send me the model, please.

**Reply by Sacropyne** [reply]
Okay ! What’s the easiest way to share a model ? I have never had an occasion to export any (note: I am using Python’s OrTools)

**Comment by lperron** [comment]
model.export_to_file('issue4310.pb.txt')

and send me the file.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le dim. 14 juil. 2024 à 16:46, Sacropyne ***@***.***> a
écrit :

**Reply by Sacropyne** [reply]
Here it is :

https://drive.google.com/file/d/1xjuIA1oHl8VV-70UT2N50Rgw-xMhJQL8/view?usp=drivesdk

Thanks !!

**Reply by Sacropyne** [reply]
Hi ! Have you been able to reproduce the same behavior on your end ?
Thanks again for your time
